ksql-engine/src/main/java/io/confluent/ksql/KsqlExecutionContext.java

```diff
@@ -100,6 +100,23 @@
    */
   ExecuteResult execute(ConfiguredStatement<?> statement);
```

Comment:
I think that is dangerous to support both `execute` methods since a small refactor will cause an issue. TBH, I think we need to do the full refactor now and make sure that _all_ service contexts are passed in (not taken from the execution context).

Reply:
This becomes interesting for this PR. I'm not sure how other commands that cannot be impersonated would behave. I was looking at how many places I need to pass the `ServiceContext`, and I found some places (like the RequestValidator) where I don't know if it requires the REST ServiceContext or the primary ServiceContext (which uses KSQL credentials).

I still need to work on other PR related to this, though, so do you think I can do it in a follow-up PR?

Reply:
Is there any way we can test that the right commands are at least getting the right service contexts?